graylog2-web-interface/src/components/sidecars/configuration-forms/ConfigurationVariablesHelper.jsx

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import { Alert, Button, Table, Modal } from 'react-bootstrap';
+import { Spinner } from 'components/common';
+import { BootstrapModalConfirm } from 'components/bootstrap';
+import BootstrapModalWrapper from 'components/bootstrap/BootstrapModalWrapper';
+import CombinedProvider from 'injection/CombinedProvider';
+import EditConfigurationVariableModal from './EditConfigurationVariableModal';
+import ConfigurationHelperStyle from './ConfigurationHelper.css';
+
+const { ConfigurationVariableActions } = CombinedProvider.get('ConfigurationVariable');
+
+class ConfigurationVariablesHelper extends React.Component {
+  state = {
+    configurationVariables: [],
+    errorModalContent: {},
+  };
+
+  componentDidMount() {
+    this._reloadConfiguration();
+  }
+
+  _reloadConfiguration = () => {
+    ConfigurationVariableActions.all()
+      .then((configurationVariables) => {
+        this.setState({ configurationVariables: configurationVariables });
+      });
+  };
+
+  _getId = (idName, index) => {
+    const idIndex = index !== undefined ? `. ${index}` : '';
+    return idName + idIndex;
+  };
+
+  _renderConfigList = (configurations) => {
+    return (<ul className={ConfigurationHelperStyle.ulStyle}>
+      {configurations.map(conf => <li key={conf.id}><a href={conf.id}>{conf.name}</a></li>)}
+    </ul>);
+  };
+
+  _configurationVariableListFormatter = () => {
+    const variableRows = [];
+
+    Object.values(this.state.configurationVariables).forEach((configVar) => {
+      let deleteConfirmationModal;
+      const _handleDeleteCheck = () => {
```

Comment:
I think it would make the code clearer moving this function and `_handleDeleteConfirm` out of the loop and into the class level. They could accept arguments with the variables they need to do their thing, and then make them return a function the consumers can call.